protocol.md

```diff
@@ -182,6 +182,29 @@ The `Tus-Max-Size` response header MUST be a non-negative integer indicating the
 allowed size of an entire upload in bytes. The Server SHOULD set this header if
 there is a known hard limit.
 
+#### Tus-Min-Chunk-Size
+
+The `Tus-Min-Chunk-Size` response header MUST be a non-negative integer indicating the
+recommended minimum size of a single chunk uploaded in a single `PATCH` request in bytes.
+The Server MAY set this recommendation if the underlying storage engine has a lower limit
+on how much data can be stored in a single take. The Client SHOULD use this number to
+determine in conjunction with other factors, such as the available bandwidth and the
+`Tus-Max-Chunk-Size` header, how much data SHOULD be transfered in one request. Furthermore,
+this header's value MAY represent a hard limit but also MAY be a recommendation. The Server
+SHOULD only set this header if it has this recommendation.
+
+#### Tus-Max-Chunk-Size
+
+The `Tus-Max-Chunk-Size` response header MUST be a non-negative integer indicating the
+recommended maximum size of a single chunk uploaded in a single `PATCH` request in bytes.
+The Server MAY set this recommendation if the underlying storage engine has an upper limit
+on how much data can be stored in a single take or if the used architecture does not allow
+streaming the body of the `PATCH` request. The Client SHOULD use this number to determine
```

Comment:
or if the used architecture does not allow the body of the 'PATCH' request to be streamed.